File: 1299-Replace-Elements-with-Greatest-Element-on-Right-Side.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        n = len(arr)
        res = [0] * n
        max = -1

        for i in range(n - 1, -1, -1):
            res[i] = max
            if arr[i] > max:
                max = arr[i]

        return res

File: neetcode/1299-Replace-Elements-with-Greatest-Element-on-Right-Side.ts
function replaceElements(arr: number[]): number[] {
  const n: number = arr.length
  const res: number[] = new Array(n).fill(-1)
  let max: number = -1

  for (let i = n - 1; i >= 0; i--) {
    res[i] = max
    if (arr[i] > max) max = arr[i]
  }

  return res
};